gnd/src/main/java/com/google/android/gnd/ui/map/gms/GoogleMapsMapAdapter.java

```diff
@@ -112,17 +116,14 @@ public void renderJsonLayer() {
           sb.append(line).append("\n");
           line = buf.readLine();
         }
-        try {
-          JSONObject geoJson = new JSONObject(sb.toString());
-          GeoJsonLayer layer = new GeoJsonLayer(map, geoJson);
-          layer.addLayerToMap();
-          Log.d(TAG, "JSON layer successfully loaded.");
-        } catch (JSONException e) {
-          Log.d(TAG, "Unable to read JSON.", e);
-        }
-      } catch (IOException e) {
-        Log.d(TAG, "GeoJSON file not found.", e);
-      }
+
+        JSONObject geoJson = new JSONObject(sb.toString());
```

Comment:
I think you can replace this with:
```java
JSONTokenizer tokenizer = new JSONTokenizer(file);
JSONObject geoJson = new JSONObject(tokenizer);

Reply:
Thank you! :tada:  I figured there was a better way to do this, but for whatever reason I had a hard time finding the API for it. I couldn't find the tokenizer, but android has a JsonReader that serves this purpose.

Reply:
Actually, The JSONReader doesn't seem to actually make anything easier/more obvious than the current impl. Let's return to this a bit later?